anvil/geometry.py

```diff
@@ -13,48 +13,68 @@ class ShiftsMismatchError(Exception):
+       identifies the position of the site (x,y) in the split representation.
+
+       splitlexi (Split to Lexicographic) is an array of integers of size
+       length*length. The element i of the array contains the index that identifies the position of the site i in the split representation, where i is the lexicographic index of the site.
     """
 
     def __init__(self, length):
         self.length = length
         # Hard code the checkerboard pattern, could be passed to class.
+        # ok for now, but needs to be changed. 
+        # the change should be a simple one
         checkerboard = torch.zeros((self.length, self.length), dtype=bool)
         checkerboard[1::2, 1::2] = True
         checkerboard[::2, ::2] = True
         self.checkerboard = checkerboard
         self.flat_checker = self.checkerboard.flatten()
         # make split-flat state like object with corresponding indices in flat state
-        self.flat_ind_like_split = torch.cat(
-            [
-                torch.where(checkerboard.flatten())[0],
-                torch.where(~checkerboard.flatten())[0],
-            ],
-            dim=0,
-        )
-        self.split_ind_like_state = self._split_indices_like_state()
+        self.splitcart = self._splitcart()
+        self.splitlexi = self._splitlexi()
 
-    def _split_indices_like_state(self):
-        """Internal function for calculating the split index like state, which
-        is the index of phi_a and phi_b in the flattened state, arranged in the
-        shape of the 2D state, used in get_shift
+    def _splitcart(self):
+        """ Internal function for calculating the splitcart grid as described above
         """
-        splitind_like_state = torch.zeros((self.length, self.length), dtype=torch.int)
-        splitind_like_state[self.checkerboard] = torch.arange(
+        lsplitcart = torch.zeros((self.length, self.length), dtype=torch.int) 
+        lsplitcart[self.checkerboard] = torch.arange(
             int(ceil(self.length ** 2 / 2)), dtype=torch.int
         )
-        splitind_like_state[~self.checkerboard] = torch.arange(
+        lsplitcart[~self.checkerboard] = torch.arange(
             int(ceil(self.length ** 2 / 2)), self.length ** 2, dtype=torch.int
         )
-        return splitind_like_state
+        return lsplitcart
+
+    def _splitlexi(self):
+        """ Internal function for calculating the partlexi array as described above
+        """
+        lsplitlexi = torch.cat(
+            [
+                torch.where(self.flat_checker)[0],
+                torch.where(~self.flat_checker)[0],
+            ],
+            dim=0,
+        )
+        return lsplitlexi
 
     def get_shift(self, shifts: tuple = (1, 1), dims: tuple = (0, 1)) -> torch.Tensor:
-        r"""Given length, which refers to size of a 2D state (length * length)
+        """Given length, which refers to size of a 2D state (length * length)
```

Comment:
r was purposely here because we have backslashes in the docstring which we explicitly don't want to treat as escape characters https://docs.python.org/3/reference/lexical_analysis.html?highlight=literals#string-and-bytes-literals

Reply:
ok. good! I thought I had added the extra 'r' by cutting and pasting stuff.